.script/utils/gitWrapper.ts

```diff
@@ -1,21 +1,35 @@
 import { cli, devOps } from "@azure/avocado";
 import * as logger from "./logger";
 import "./stringExtenssions";
+import { PullRequestProperties } from '@azure/avocado/dist/dev-ops';
 
-export async function GetDiffFiles(fileTypeSuffixes?: string[], filePathFolderPreffixes?: string[]) {
+var pullRequestDetails: PullRequestProperties | undefined;
+var isPullRequestDetailsInitialized: boolean = false;
+
+export async function GetPRDetails() {
+  if (!isPullRequestDetailsInitialized){
+    console.log("Getting PR details");
+    const config = cli.defaultConfig();
+    pullRequestDetails = await devOps.createPullRequestProperties(config);
+    isPullRequestDetailsInitialized = true;
+  }
+  return pullRequestDetails;
+}
+
+export async function GetDiffFiles(fileTypeSuffixes?: string[], filePathFolderPreffixes?: string[], fileKinds?: string[]) {
   const config = cli.defaultConfig();
   const pr = await devOps.createPullRequestProperties(config);
 
   if (typeof pr === "undefined") {
     console.log("Azure DevOps CI for a Pull Request wasn't found. If issue persists - please open an issue");
     return;
   }
-
+ 
   let changedFiles = await pr.diff();
   console.log(`${changedFiles.length} files changed in current PR`);
 
   const filterChangedFiles = changedFiles
-    .filter(change => change.kind !== "Deleted")
+    .filter(change => fileKinds?.includes(change.kind))
```

Comment:
In case that fileKinds is empty, what will return after the filter?

I think of 2 options, maybe the secone one is more generic:
 -  use this filter: change.kind !== "Deleted"
 - returns all files. Added, Modified and deleted

Reply:
Well, in my case I need to validate just the Modified files. 
I think it's more generic to let anyone who wants to use this function to decide what kinds of files to filter by.
Currently, In case that fileKinds is empty, it will return an empty list.

Therefore I consider define it as a mandatory parameter and not optional, WDYT?